internal/provider/data_source_packer_image.go

```diff
@@ -0,0 +1,226 @@
+package provider
+
+import (
+	"context"
+	"log"
+	"time"
+
+	packermodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-packer-service/preview/2021-04-30/models"
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+)
+
+// a minute sounds like a lot already since this would be mainly pulling data.
+var defaultPackerTimeout = time.Minute
+
+func dataSourcePackerImage() *schema.Resource {
+	return &schema.Resource{
+		Description: "The Packer Image data source provides information about an existing image build stored in the Packer registry",
+		ReadContext: dataSourcePackerImageRead,
+		Timeouts: &schema.ResourceTimeout{
+			Default: &defaultPackerTimeout,
+		},
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"bucket": {
+				Description:      "The ID of the HCP Packer Registry image bucket to pull from.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			// Optional inputs
+			"channel": {
+				Description:      "The channel that points to the version of the image you want.",
+				Type:             schema.TypeString,
+				Optional:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			// computed outputs
+			"organization_id": {
+				Description: "The ID of the organization this HCP Packer registry is located in.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"project_id": {
+				Description: "The ID of the project this HCP Packer registry is located in.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+
+			"incremental_version": {
+				Description: "The Packer version of the registry.",
+				Type:        schema.TypeInt,
+				Computed:    true,
+			},
+			"created_at": {
+				Description: "The time that the Packer registry was created.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"image_id": {
+				Description: "The ID of the image.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"builds": {
```

Comment:
Would it be possible to flatten this object at all?

Reply:
Its definite format will evolve for a bit. Almost there. I'll try to flatten it as much as I can. 🙂

Reply:
It's done, tell me what you think; this one is very close to the schema of our API. 🙂